cmd/pass-through-nse/pass_through_connect.go

```diff
@@ -0,0 +1,335 @@
+	// remove from connections
+	logrus.Infof("passThrough DeleteConnection: %+v", conn)
+
+	if endpoint.Next(ctx) != nil {
+		return endpoint.Next(ctx).Close(ctx, conn)
+	}
+	return &empty.Empty{}, nil
+}
+
+//processNsEndpoints finds the label of the chain endpoint and sends a NS request to connect with it
+func (ptxc *passThroughComposite) processNsEndpoints(ctx context.Context, response *registry.FindNetworkServiceResponse, remoteIP string,
+										logger *logrus.Logger) (*connection.Connection, error) {
+	logger.Infof("Processing NS endpoints...")
+
+	chainLabelKey, chainLabelVal, err := ptxc.findChainEndpointLabel(response, logger)
+
+	if err != nil {
+		return nil, err
+	}
+
+	//routes := []string{ptxc.passThroughNetCidr}
+	chainConn, connErr := ptxc.createChainConnection(ctx, response, logger, remoteIP, chainLabelKey, chainLabelVal)
+
+	if connErr != nil {
+		return nil, connErr
+	}
+
+	return chainConn, nil
+}
+
+// findChainEndpointLabel finds the label for the chain endpoint
+func(ptxc *passThroughComposite) findChainEndpointLabel(response *registry.FindNetworkServiceResponse, logger *logrus.Logger) (string, string, error) {
+	var chainLabelKey, chainLabelVal string
+
+	// There must be >= 2 matches under the networkservice to perform NSE chaining:
+	// 1 match for current endpoint, the other match(es) for the chaining NSE
+	matches := response.GetNetworkService().GetMatches()
+	if len(matches) < 2 {
+		return "", "", fmt.Errorf("Failed to find chain NSE: current networkservice has %d match", len(matches))
+	}
+
+	logger.Infof("The matches under current networkservice are: %v", matches)
+	// An endpoint has at least 1 label for it to be found by NSMgr
+	for myLabelKey, myLabelVal := range ptxc.myEndpointLabels{
+		for _, match := range matches {
+			// First find the match that has the same label as the current endpoint
+			if val, ok := match.SourceSelector[myLabelKey]; ok && val == myLabelVal {
+				// There should be only one route and one destination since one client only connects to one endpoint under the same networkservice
+				if len(match.GetRoutes()) == 0{
+					return "", "", fmt.Errorf("Failed to find chain NSE label -- no route found under current match")
+				}
+				if (len(match.GetRoutes())) > 1 {
+					return "", "", fmt.Errorf("Failed to find chain NSE label -- more than 1 routes found under current match")
+				}
+
+				if len(match.GetRoutes()[0].GetDestinationSelector()) == 0 {
+					return "", "", fmt.Errorf("Failed to find chain NSE label -- no destination found under current route")
+				}
+				if len(match.GetRoutes()[0].GetDestinationSelector()) > 1 {
+					return "", "", fmt.Errorf("Failed to find chain NSE label -- more than 1 destinations found under current route")
+				}
+
+				// Get the destination label key and value
+				for chainLabelKey, chainLabelVal = range match.GetRoutes()[0].GetDestinationSelector() {}
+
+				break
```

Comment:
Yea.. line 163 is to get the only entry in the `DestinationSelector`, is there another way in Golang to get the only entry on the map?

In terms of the multiple key-values in the `DestinationSelector`, this is not possible IMO because we only allow one pass-through to chain with one vl3